src/main/kotlin/uk/gov/justice/digital/hmpps/hmppspersononprobationuserapi/service/DelegatedAccessService.kt

```diff
@@ -0,0 +1,126 @@
+package uk.gov.justice.digital.hmpps.hmppspersononprobationuserapi.service
+
+import jakarta.transaction.Transactional
+import jakarta.validation.ValidationException
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppspersononprobationuserapi.config.DuplicateDataFoundException
+import uk.gov.justice.digital.hmpps.hmppspersononprobationuserapi.config.ResourceNotFoundException
+import uk.gov.justice.digital.hmpps.hmppspersononprobationuserapi.data.DelegatedAccess
+import uk.gov.justice.digital.hmpps.hmppspersononprobationuserapi.jpa.entity.DelegatedAccessEntity
+import uk.gov.justice.digital.hmpps.hmppspersononprobationuserapi.jpa.entity.DelegatedAccessPermissionEntity
+import uk.gov.justice.digital.hmpps.hmppspersononprobationuserapi.jpa.repository.DelegatedAccessPermissionRepository
+import uk.gov.justice.digital.hmpps.hmppspersononprobationuserapi.jpa.repository.DelegatedAccessRepository
+import uk.gov.justice.digital.hmpps.hmppspersononprobationuserapi.jpa.repository.UserRepository
+import java.time.LocalDateTime
+
+@Service
+class DelegatedAccessService(private val delegatedAccessRepository: DelegatedAccessRepository, private val userRepository: UserRepository, private val delegatedAccessPermissionRepository: DelegatedAccessPermissionRepository) {
+
+  @Transactional
+  fun createDelegatedAccess(delegatePost: DelegatedAccess): DelegatedAccessEntity {
+    val now = LocalDateTime.now()
+
+    val initiatedUserExists = userRepository.existsById(delegatePost.initiatedUserId.toLong()) ?: throw ResourceNotFoundException("User with id ${delegatePost.initiatedUserId} not found in database ")
+    val delegatedUserExists = userRepository.existsById(delegatePost.delegatedUserId.toLong()) ?: throw ResourceNotFoundException("User with id ${delegatePost.delegatedUserId} not found in database ")
```

Comment:
The call to exists by id should always be returning true or false - what are the nullable defaults fall?

Reply:
By default it will have value  0 if not provided in the request

Reply:
this check is provided in the code.